File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/context2.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import ChatWithContextPostRequest, ContextCreateRequest, ContextResponse

class Context2Api:
    """Context API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def chat_with(self, contextId: str, body: ChatWithContextPostRequest) -> ContextResponse:
        """Chat with context"""
        return self._client.post(f"/ai/v3/v1/context/{contextId}/chat/completions", json=body)

    def create_context(self, body: ContextCreateRequest) -> ContextResponse:
        """Create context"""
        return self._client.post(f"/ai/v3/v1/context/create", json=body)

    def get_context(self, contextId: str) -> ContextResponse:
        """Get context"""
        return self._client.get(f"/ai/v3/v1/context/{contextId}")

    def delete_context(self, contextId: str) -> None:
        """Delete context"""
        return self._client.delete(f"/ai/v3/v1/context/{contextId}")

File: sdkwork-ai-sdk-flutter/lib/src/api/context2.dart
import '../http/client.dart';
import '../models.dart';

class Context2Api {
  final HttpClient _client;
  
  Context2Api(this._client);

  /// Chat with context
  Future<ContextResponse?> chatWith(String contextId, ChatWithContextPostRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/v1/context/${contextId}/chat/completions'), body: body, contentType: 'application/json');
    return response is ContextResponse ? response : null;
  }

  /// Create context
  Future<ContextResponse?> createContext(ContextCreateRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/v1/context/create'), body: body, contentType: 'application/json');
    return response is ContextResponse ? response : null;
  }

  /// Get context
  Future<ContextResponse?> getContext(String contextId) async {
    final response = await _client.get(ApiPaths.aiPath('/v1/context/${contextId}'));
    return response is ContextResponse ? response : null;
  }

  /// Delete context
  Future<void> deleteContext(String contextId) async {
    await _client.delete(ApiPaths.aiPath('/v1/context/${contextId}'));
  }
}
